src/main/c/Jep/convert_p2j.c

```diff
@@ -55,18 +55,25 @@ static void raiseTypeError(JNIEnv *env, PyObject *pyobject, jclass expectedType)
     if (process_java_exception(env)) {
         return;
     }
+    PyObject *name_object = NULL;
     expTypeName = (*env)->GetStringUTFChars(env, expTypeJavaName, 0);
     if (PyJClass_Check(pyobject)) {
         actTypeName = "java.lang.Class";
     } else if (PyJObject_Check(pyobject)) {
         actTypeName = PyUnicode_AsUTF8(((PyJObject*) pyobject)->javaClassName);
     } else {
+#ifdef Py_LIMITED_API
+        name_object = PyObject_Str((PyObject*) Py_TYPE(pyobject));
+        actTypeName = PyUnicode_AsUTF8(name_object);
+#else
         actTypeName = pyobject->ob_type->tp_name;
```

Comment:
So there's no way to get `PyTypeObject->tp_name` in the Stable ABI, and I just approximate it as a sequence of Python-level calls.....